crates/c-api/include/wasi.h

```diff
@@ -39,6 +39,45 @@ extern "C" {
  */
 WASI_DECLARE_OWN(config)
 
+/**
+ * \typedef wasi_read_pipe_t
+ * \brief Convenience alias for #wasi_read_pipe_t
+ *
+ * \struct wasi_read_pipe_t
+ * \brief Reader end of a bounded FIFO buffer
+ * 
+ * An instance using fd_read on an empty pipe with succeeded but with 0 bytes read.
+ * The effect depends on the implementation. For example, Rust's
+ * `std::io::Read::read` returns Ok(0), while `read_exact` returns Err
```

Comment:
I'm not sure what this is intending to communicate? Could you clarify what this is explaining?

Reply:
Wasi fd_read (like POSIX one), returns the number of byte read.
Then the queue is empty, the returned value is zero.
Effects on programs may varies.
For example, a c program without checks will just ignore this.
While rust print macro unwraps `std::io::Read::read_exact` and so panic.

If you have a good way to explain that I'm taking it…